packages/assets-controllers/src/TokensController.ts

```diff
@@ -263,34 +270,55 @@ export class TokensController extends BaseController<
     });
   }
 
-  _instantiateNewEthersProvider(): any {
-    return new Web3Provider(this.config?.provider);
+  /**
+   * Get the relevant provider instance.
+   *
+   * @param networkClientId - Network Client ID.
+   * @returns Web3Provider instance.
+   */
+  #getProvider(networkClientId?: NetworkClientId): Web3Provider {
+    const provider = networkClientId
+      ? this.getNetworkClientById(networkClientId).provider
+      : this.config?.provider;
+
+    return new Web3Provider(provider);
   }
 
   /**
    * Adds a token to the stored token list.
    *
-   * @param address - Hex address of the token contract.
-   * @param symbol - Symbol of the token.
-   * @param decimals - Number of decimals the token uses.
-   * @param options - Object containing name and image of the token
-   * @param options.name - Name of the token
-   * @param options.image - Image of the token
-   * @param options.interactingAddress - The address of the account to add a token to.
+   * @param args - The method argument object.
+   * @param args.address - Hex address of the token contract.
+   * @param args.symbol - Symbol of the token.
+   * @param args.decimals - Number of decimals the token uses.
+   * @param args.options - Object containing name and image of the token
+   * @param args.options.name - Name of the token
+   * @param args.options.image - Image of the token
+   * @param args.options.interactingAddress - The address of the account to add a token to.
+   * @param args.networkClientId - Network Client ID.
    * @returns Current token list.
    */
-  async addToken(
-    address: string,
-    symbol: string,
-    decimals: number,
-    {
-      name,
-      image,
-      interactingAddress,
-    }: { name?: string; image?: string; interactingAddress?: string } = {},
-  ): Promise<Token[]> {
+  async addToken({
+    address,
+    symbol,
+    decimals,
+    options: { name, image, interactingAddress } = {},
```

Comment:
I wonder if we just remove this options object since all the args are now in an object so we can just make them optional that way... Otherwise this becomes a strange function signature.